modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -591,6 +616,85 @@ private String _getObjectEntryPermissionName(long objectDefinitionId) {
 		return ObjectDefinition.class.getName() + "#" + objectDefinitionId;
 	}
 
+	private boolean _hasPreventDeletionType(long objectDefinitionId)
+		throws PortalException {
+
+		_objectRelationships = new ArrayList<>();
+
+		_objectRelationships.addAll(
+			TransformUtil.transform(
+				_objectRelationshipLocalService.getObjectRelationships(
+					objectDefinitionId),
+				objectRelationship -> {
+					if (objectRelationship.isReverse()) {
+						return null;
+					}
+
+					return objectRelationship;
+				}));
+
+		_objectRelationships.addAll(
+			_getManyToManyObjectRelationships(objectDefinitionId));
+
+		for (ObjectRelationship objectRelationship : _objectRelationships) {
+			if (StringUtil.equals(
+					objectRelationship.getDeletionType(),
+					ObjectRelationshipConstants.DELETION_TYPE_PREVENT)) {
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private boolean _hasRelatedObjectEntries(
+			com.liferay.object.model.ObjectEntry objectEntry)
+		throws PortalException {
+
+		for (ObjectRelationship objectRelationship : _objectRelationships) {
```

Comment:
@marcelabc can we try to follow the same pattern you did for a similar use case? That one that we've consulted the DB directly?